### PR TITLE
[#186510319] Add Actions support

### PR DESCRIPTION
**Pull request by selectiveduplicate**
## Test results
image

### `sendAction` test

image

image

image

image

### `sendActionsBatch` tests

image

image

image

image

### CODE REVIEW

**Comment by praves77** [comment]
Cc: @xinghengwang ^^
@selectiveduplicate 
- don't we bump up the version to publish?
- please check/include dependabot PR too.

**Comment by selectiveduplicate** [comment]
@praves77 Done! Included terminal output for local tests in the description as well.

File: lib/Models/ActionModel.js [hunk 1]
```diff
@@ -0,0 +1,118 @@
...
+    this.companyId = obj.companyId;
+    this.metadata = obj.metadata;
+    this.request = obj.request;
+  }
+
+  // Append to variable dictionary
+  this._variableDict["userId"] = "user_id";
+  this._variableDict["companyId"] = "company_id";
+  this._variableDict["sessionToken"] = "session_token";
+};
```
Comment (xinghengwang) [inline lib/Models/ActionModel.js:41]: are you sure `user_agent_string` and `ip_address` doesn't need to be in the dictionary here? for handling translating form camel case to snake case?   Is it carried over in the payload without?

**Review by xinghengwang — CHANGES_REQUESTED** [review]

File: lib/Models/ActionModel.js (hunk 1 above)
Comment (xinghengwang) [reply lib/Models/ActionModel.js:41]: I don't remember if it is is the right place to add it, just double check the { request.userAgentString  and request.ipAddress is translated correctly snake_case for the final payload.

**Comment by praves77** [comment]
@selectiveduplicate please review the @xinghengwang comments and apply.

**Comment by selectiveduplicate** [comment]
@xinghengwang You're right! They weren't being properly formatted. I've patched an implementation; let me know if there's a better way.
cc @praves77 .

**Review by xinghengwang — APPROVED** [review]

**Review by xinghengwang — APPROVED** [review]

**Pull request merged**